File: web/src/types/fluxocaixa/RegraExtratoContaCorrente.ts
import { Categoria } from "../Categoria";
import { AtivoCategoriaEnum } from "./AtivoCategoriaEnum";
import { TipoRegraExtratoContaCorrenteEnum } from "./TipoRegraExtratoContaCorrenteEnum";

export type RegraExtratoContaCorrente = {
  id: number;
  tipoRegra: TipoRegraExtratoContaCorrenteEnum;
  descricao: string;
  descricaoMatch: string;
  descricaoDestino?: string;
  despesaCategoriaDestino?: Categoria;
  rendaCategoriaDestino?: Categoria;
  ativoCategoriaDestino?: AtivoCategoriaEnum;
  prioridade: number;
  ativo: boolean;
}

export const initialRegraExtratoContaCorrenteState: RegraExtratoContaCorrente = {
  id: 0,
  tipoRegra: "CLASSIFICAR_DESPESA",
  descricao: "",
  descricaoMatch: "",
  prioridade: 0,
  ativo: true,
};

File: server/src/main/java/br/com/lcano/centraldecontrole/domain/fluxocaixa/RegraExtratoContaCorrente.java
package br.com.lcano.centraldecontrole.domain.fluxocaixa;

import br.com.lcano.centraldecontrole.domain.Usuario;
import br.com.lcano.centraldecontrole.enums.fluxocaixa.TipoRegraExtratoContaCorrente;
import br.com.lcano.centraldecontrole.util.BooleanToCharConverter;
import jakarta.persistence.*;
import lombok.Data;

@Table(name = "regraextratocontacorrente")
@Entity
@Data
public class RegraExtratoContaCorrente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "idusuario", nullable = false)
    private Usuario usuario;

    @Enumerated(EnumType.STRING)
    @Column(name = "tiporegra", nullable = false)
    private TipoRegraExtratoContaCorrente tipoRegra;

    @Column(nullable = false)
    private String descricao;

    @Column(name = "descricaomatch", nullable = false)
    private String descricaoMatch;

    @Column(name = "descricaodestino")
    private String descricaoDestino;

    @ManyToOne
    @JoinColumn(name = "idcategoriadespesadestino")
    private DespesaCategoria despesaCategoriaDestino;

    @ManyToOne
    @JoinColumn(name = "idcategoriarendadestino")
    private RendaCategoria rendaCategoriaDestino;

    @Column(nullable = false)
    private Long prioridade;

    @Convert(converter = BooleanToCharConverter.class)
    @Column(nullable = false)
    private boolean ativo;
}
